src/Streamly/Internal/LZ4.hs

```diff
@@ -213,8 +237,14 @@ compressChunk cfg speed ctx arr = do
               let uncompLen = Array.byteLength arr
                   maxUncompLen = cIntToInt lz4_MAX_INPUT_SIZE
                   speedC = unsafeIntToCInt speed
+                  maxBlockSize = blockSizeToInt $ blockSize cfg
               -- Ideally the maxCLen check below covers this case, but just in
               -- case.
+              when (uncompLen > maxBlockSize)
+                $ error $ "compressChunk: Source array length "
+                    ++ show uncompLen
+                    ++ " exceeds the max block size in the config "
+                    ++ show maxBlockSize
               when (uncompLen > maxUncompLen)
```

Comment:
We can restrict the maxBlockSize to maxUncompLen, so this check would be covered in the previous check itself.